python/taichi/lang/_ndrange.py

```diff
@@ -139,10 +139,12 @@ def ndrange(*args):
     return _Ndrange(*args)
 
 
-class GroupedNDRange:
+class GroupedNDRangeBase:
     def __init__(self, r):
         self.r = r
 
+
+class GroupedNDRange(GroupedNDRangeBase):
     def __iter__(self):
         for ind in self.r:
             if impl.current_cfg().real_matrix:
```

Comment:
IIUC the `GroupedNDRange` class is used only in `ti.static`. So it seems that removing the `real_matrix` case here is already enough and no other modification is needed.